### PR TITLE
Added Graph Types: Vertex, Edge, Property, VertexProperty, Path

### PR DESCRIPTION
**Pull request by L0Lmaker**
*Issue #, if available:*

https://bitquill.atlassian.net/jira/software/c/projects/AN/boards/36?modal=detail&selectedIssue=AN-1008

*Description of changes:*

Added Support for Graph Types to Serializers.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: gremlin-go/driver/graphBinary.go [hunk 1]
```diff
@@ -288,6 +293,200 @@ func bigIntReader(buffer *bytes.Buffer, _ *graphBinaryTypeSerializer) (interface
...
+	// read null byte
+	_, _ = typeSerializer.read(buffer)
+	_, _ = typeSerializer.read(buffer)
+	return vp, nil
+}
+
+//Format: {labels}{objects}
+func pathWriter(value interface{}, buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) ([]byte, error) {
+	p := value.(*Path)
+	_, err := typeSerializer.write(p.labels, buffer)
```
Comment (xiazcy) [inline gremlin-go/driver/graphBinary.go:462]: Note that the official definition for `Path.labels` should be a list containing sets of string, so I think we'll need to have that structure here for the server to work. 
While we are leaving actual `set` implementations for later, we can add a `set` datatype here and use `listWriter`/`listReader` for it for now. This does mean that we'll need to change the `Path` struct in `graph.go` first to have `labels [][]string` instead, then propagate that change to this in the pathWriter and pathReader.

**Review by xiazcy — CHANGES_REQUESTED** [review]
I think we should have the `Path.labels` properly structured (i.e. a list of a set, or a list of a list in our case), since that's what the server will be expecting.

**Review by lyndonbauto — APPROVED** [review]

File: gremlin-go/driver/graph.go [hunk 2]
```diff
@@ -74,7 +74,7 @@ type Property struct {
 // A list of labels and a list of objects is maintained in the path.
 // The list of labels are the labels of the steps traversed, and the objects are the objects that are traversed.
 type Path struct {
```
Comment (xiazcy) [inline gremlin-go/driver/graph.go:77]:
```suggestion
// TODO change labels to be []<set of string> after implementing set in AN-1022 and update the GetPathObject accordingly
type Path struct {
```
Adding a TODO here as a reminder

**Review by xiazcy — APPROVED** [review]
Just add the TODO regarding the Path.labels struct (you can just click commit suggestion)

**Pull request merged**